packages/x-date-pickers/src/PickersTextField/PickersOutlinedInput/PickersOutlinedInput.tsx

```diff
@@ -57,19 +55,37 @@ const PickersOutlinedInputRoot = styled(PickersInputBaseRoot, {
     [`&.${pickersOutlinedInputClasses.error} .${pickersOutlinedInputClasses.notchedOutline}`]: {
       borderColor: (theme.vars || theme).palette.error.main,
     },
+    variants: Object.keys((theme.vars ?? theme).palette)
```

Comment:
Not sure if this is the place to blame, but it looks like after the change the specificity is a bit "wrong". 🙈 

https://github.com/mui/mui-x/assets/4941090/7076ac29-4599-49ca-8a0a-261525adc950

Reply:
Yep, did not look into this one yet :+1:

Reply:
Awesome that you found this one :grimacing: 

This should be fixed :ok_hand: